examples/fileuploader.hs

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env stack
+-- stack --resolver lts-6.27 runghc --package minio-hs --package optparse-applicative --package filepath
+
+{-# Language OverloadedStrings #-}
+import Network.Minio
+
+import Control.Monad.Catch (catch)
+import Control.Monad.IO.Class (liftIO)
+import Options.Applicative
+import Prelude
+import System.FilePath.Posix
+import Data.Text (pack)
+
+-- | The following example uses minio's play server at
+-- https://play.minio.io:9000.  The endpoint and associated
+-- credentials are provided via the libary constant,
+--
+-- > minioPlayCI :: ConnectInfo
+--
+
+-- optparse-applicative package based command-line parsing.
+fileNameOpts :: Parser FilePath
+fileNameOpts = strOption
+               (long "filename"
+                <> metavar "FILENAME"
+                <> help "Name of file to upload to AWS S3 or a Minio server")
+
+cmdParser = info
+            (helper <*> fileNameOpts)
+            (fullDesc
+             <> progDesc "fileUploader"
+             <> header
+             "fileUploader - a simple file-uploader program using minio-hs")
+
+
+main :: IO ()
+main = do
+  let bucket = "my-bucket"
+
+  -- Parse command line argument, namely --filename.
+  filepath <- execParser cmdParser
+  let object = pack $ takeBaseName filepath
+
+  res <- runResourceT $ runMinio minioPlayCI $ do
+    -- Make a bucket; catch bucket already exists exception if thrown.
+    (catch :: Minio () -> (MError -> Minio ()) -> Minio ())
+      (makeBucket bucket Nothing)
+      (\_ -> liftIO $ print "Bucket already exists, proceeding with upload file.")
+
```

Comment:
After adding ScopedTypeVariables, we can simply do:

```haskell
    catch
      (makeBucket bucket Nothing)
      (\(_ :: MError) -> liftIO $
        putStrLn "Bucket already exists, proceeding with upload file.")
```

I think looks more readable.